### PR TITLE
Support negative index for Gather op

### PR DESCRIPTION
**Pull request by duli2012**
**Description**: Describe your changes.
Adding support for negative index in Gather op

**Motivation and Context**
- Why is this change required? What problem does it solve?
Gather spec allows negative index, however, our current implementation doesn't support it, which blocked a custom model converted from pytorch.

- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/gather.cc [hunk 1]
```diff
@@ -41,7 +41,7 @@ Status GatherCopyData(const Tensor* indices_tensor, const uint8_t* src_base, uin
   // We can't merge this code in the omp loop below as omp does not allow return in the loop
   for (int64_t i = 0; i < N; ++i) {
     Tin idx = indices_data[i];
-    if (idx < 0 || idx >= input_data_shape[axis]) {
+    if (idx < -input_data_shape[axis] || idx >= input_data_shape[axis]) {
```
Comment (skottmckay) [inline onnxruntime/core/providers/cpu/tensor/gather.cc]: Can you use HandleNegativeAxis in providers/common.h?

Comment (duli2012) [reply onnxruntime/core/providers/cpu/tensor/gather.cc]: Good point! Done.

Comment (tracysh) [reply onnxruntime/core/providers/cpu/tensor/gather.cc]: Does this exception from HandleNegativeAxis (with its exception text confusingly referring to axis not indices) work properly across what is currently the OMP boundary?

Comment (duli2012) [reply onnxruntime/core/providers/cpu/tensor/gather.cc]: @tracysh are you suggesting to change the exception text?

**Comment by duli2012** [comment]
After internal discussions, we feel the spec has to add clarification before this runtime change, so we hold it off now.

**Pull request closed without merging**